File: test/data/cs/test04/StateEscaped.cs
public class StateEscaped : BaseState
{
    private StateController _stateController; 
    private IControllee _controllee; 
    public StateEscaped(StateController stateController, IControllee controllee) : base(controllee)
    {
        _stateController = stateController;
        _controllee = controllee;
    }
    protected override string GetStateName()
    {
        return "Escaped"; 
    }
}

File: test/data/cpp/test03/StateEscaped.cpp
#include "StateEscaped.h"
StateEscaped::StateEscaped(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateEscaped::~StateEscaped()
{
}
const char* StateEscaped::GetStateName()
{
    return "Escaped";
}
BaseState* StateEscaped::GetParent()
{
    return nullptr;
}
int StateEscaped::GetStateID()
{
    return 1;
}